executor/common_fuchsia.h

```diff
@@ -72,24 +72,28 @@ static void* ex_handler(void* arg)
 			continue;
 		}
 		debug("got exception packet: type=%d status=%d tid=%llu\n",
-		      packet.type, packet.status, packet.exception.tid);
+		      packet.type, packet.status, static_cast<unsigned long long>(packet.exception.tid));
 		zx_handle_t thread;
 		status = zx_object_get_child(zx_process_self(), packet.exception.tid,
 					     ZX_RIGHT_SAME_RIGHTS, &thread);
 		if (status != ZX_OK) {
 			debug("zx_object_get_child failed: %d\n", status);
 			continue;
 		}
-		uint32 bytes_read;
-		zx_x86_64_general_regs_t regs;
-		status = zx_thread_read_state(thread, ZX_THREAD_STATE_REGSET0,
-					      &regs, sizeof(regs), &bytes_read);
-		if (status != ZX_OK || bytes_read != sizeof(regs)) {
-			debug("zx_thread_read_state failed: %d/%d (%d)\n",
-			      bytes_read, (int)sizeof(regs), status);
+		zx_thread_state_general_regs_t regs;
+		status = zx_thread_read_state(thread, ZX_THREAD_STATE_GENERAL_REGS,
+					      &regs, sizeof(regs));
+		if (status != ZX_OK) {
+			debug("zx_thread_read_state failed: %d (%d)\n",
+			      (int)sizeof(regs), status);
 		} else {
+			#if defined(__x86_64__) || 0
```

Comment:
Strictly saying, # must be the first character of a line. And || 0 is unnecessary.
Please do something like:
```
#if defined(__x86_64__)
        ...
#elif defined(__aarch64__)
        ...
#else
#error "unsupported arch"
#endif

Reply:
Fixed.

Out of curiosity, is there ever a reason for the "|| 0" in this sort of design? I was mimicking the pattern of doing this, established in syscalls_akaros.h, syscalls_fuchsia.h, syscalls_linux.h?

Reply:
Ah, that one :)
That's an autogenerated file. And it's generated with Go text templates, and it was easier to generate "ARCH ||" for each arch define (some arches have more than 1), and then this ends up with || at the end, so I just appended 0 and the end of line.
Ideally we fix these ||0, but it just was never too important.